### PR TITLE
Implement args builder for selectObjectContent API

### PR DESCRIPTION
**Pull request by anjalshireesh**
Introduced new args class `SelectObjectContentArgs` that extends from
`SsecObjectArgs` and has the following attributes:
  - bucket (inherited)
  - object (inherited)
  - ssec (inherited)
  - sqlExpression
  - inputSerialization
  - outputSerialization
  - requestProgress
  - scanStartRange
  - scanEndRange

### CODE REVIEW

File: api/src/main/java/io/minio/SelectObjectContentArgs.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    return sqlExpression;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder
+      extends SsecObjectArgs.Builder<Builder, SelectObjectContentArgs> {
+    public Builder sqlExpression(String sqlExpression) {
```
Comment (balamurugana) [inline api/src/main/java/io/minio/SelectObjectContentArgs.java:61]: We would need to validate `sqlExpression` is not null and not empty.

File: api/src/main/java/io/minio/SelectObjectContentArgs.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      operations.add(args -> args.sqlExpression = sqlExpression);
+      return this;
+    }
+
+    public Builder inputSerialization(InputSerialization inputSerialization) {
```
Comment (balamurugana) [inline api/src/main/java/io/minio/SelectObjectContentArgs.java:66]: `inputSerialization` can be validated to non-null.

File: api/src/main/java/io/minio/SelectObjectContentArgs.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      operations.add(args -> args.inputSerialization = inputSerialization);
+      return this;
+    }
+
+    public Builder outputSerialization(OutputSerialization outputSerialization) {
```
Comment (balamurugana) [inline api/src/main/java/io/minio/SelectObjectContentArgs.java:71]: `outputSerialization` can be validated to non-null.

Also we need to have `validate()` method to validate `sqlExpression`, `inputSerialization` and `outputSerialization`

File: docs/API.md [hunk 4]
```diff
@@ -1580,24 +1580,16 @@ for (Result<DeleteError> result : results) {
...
-| ``bucketName``      | _String_                            | Name of the bucket.                   |
-| ``objectName``      | _String_                            | Object name in the bucket.            |
-| ``sqlExpression``   | _String_                            | SQL expression.                       |
-| ``is``              | _[InputSerialization]_              | Input specification of object data.   |
-| ``os``              | _[OutputSerialization]_             | Output specification of result.       |
-| ``requestProgress`` | _boolean_                           | Flag to request progress information. |
-| ``scanStartRange``  | _Long_                              | scan start range of the object.       |
-| ``scanEndRange``    | _Long_                              | scan end range of the object.         |
-| ``sse``             | _[ServerSideEncryptionCustomerKey]_ | SSE-C type server-side encryption.    |
+| ``args``            | _SelectObjectContentArgs_           | Arguments.                            |
```
Comment (balamurugana) [inline docs/API.md:1592]: `SelectObjectContentArgs` requires `[]` around to make it as link.

File: functional/FunctionalTest.java [hunk 5]
```diff
@@ -4287,11 +4293,11 @@ public static void selectObjectContent_test1() throws Exception {
             "stats.bytesReturned mismatch; expected: 222, got: " + stats.bytesReturned());
       }
 
-      mintSuccessLog(testName, null, startTime);
+      mintSuccessLog(methodName, args, startTime);
     } catch (Exception e) {
       mintFailedLog(
```
Comment (balamurugana) [inline functional/FunctionalTest.java:4298]: We can use `handleException()` here.

File: api/src/main/java/io/minio/SelectObjectContentArgs.java (hunk 1 above)
Comment (anjalshireesh) [reply api/src/main/java/io/minio/SelectObjectContentArgs.java:61]: Will add the validation

File: api/src/main/java/io/minio/SelectObjectContentArgs.java (hunk 2 above)
Comment (anjalshireesh) [reply api/src/main/java/io/minio/SelectObjectContentArgs.java:66]: Will add the validation

File: api/src/main/java/io/minio/SelectObjectContentArgs.java (hunk 3 above)
Comment (anjalshireesh) [reply api/src/main/java/io/minio/SelectObjectContentArgs.java:71]: Will add

File: docs/API.md (hunk 4 above)
Comment (anjalshireesh) [reply docs/API.md:1592]: My bad, thanks!

File: functional/FunctionalTest.java (hunk 5 above)
Comment (anjalshireesh) [reply functional/FunctionalTest.java:4298]: Will do

File: api/src/main/java/io/minio/SelectObjectContentArgs.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage,
+ * (C) 2015, 2016, 2017, 2018, 2019 MinIO, Inc.
```
Comment (kannappanr) [inline api/src/main/java/io/minio/SelectObjectContentArgs.java:3]:
```suggestion
 * (C) 2020 MinIO, Inc.
```

Comment (anjalshireesh) [reply api/src/main/java/io/minio/SelectObjectContentArgs.java:3]: Will change

**Pull request merged**